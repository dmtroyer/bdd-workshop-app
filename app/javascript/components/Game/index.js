import React from 'react';

const Game = () => (
  <section className="main-content">DO YOU WANT TO PLAY A GAME_</section>
);

export default Game;
