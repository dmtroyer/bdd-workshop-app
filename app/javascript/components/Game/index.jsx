import React from 'react';
import { Redirect } from 'react-router-dom';
import { Spinner } from 'reactstrap';
import useProfile from '../../hooks/useProfile';
import MainContainer from '../MainContainer';
import GameProfile from '../Profile/GameProfile';

const Game = () => {
  const { profile, loading } = useProfile();

  if (loading) return <Spinner data-testid="game-spinner" />;
  if (profile === undefined) {
    return <Redirect to="/register" />;
  }
  return (
    <MainContainer dataTestId="game">
      <h1>coming soon</h1>
      <GameProfile profile={profile} />
    </MainContainer>
  );
};

export default Game;
