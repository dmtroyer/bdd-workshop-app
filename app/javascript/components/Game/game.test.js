import React from 'react';
import { shallow } from 'enzyme';
import Game from '.';

describe('Game', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Game />);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a <div>', () => {
    expect(wrapper.type()).toBe('div');
  });

  it('renders a div with message and one with version as children', () => {
    expect(wrapper.find('[data-test="message"]').prop('children')).toBe(
      'You are on React'
    );
    expect(wrapper.find('[data-test="react-version"]').prop('children')).toBe(
      '16.8.4'
    );
  });
});
