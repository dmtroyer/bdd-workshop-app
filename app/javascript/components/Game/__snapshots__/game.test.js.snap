// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game renders correctly 1`] = `
<section
  className="main-content"
>
  DO YOU WANT TO PLAY A GAME_
</section>
`;
